src/dropdown/dropdown.spec.ts

```diff
@@ -361,44 +374,47 @@ describe('ngb-dropdown-toggle', () => {
     expect(compiled).toBeShown();
   });
 
-  it('should close on ESC', () => {
-    const html = `
-      <div ngbDropdown>
-          <button ngbDropdownToggle>Toggle dropdown</button>
-          <div ngbDropdownMenu></div>
-      </div>`;
-
-    const fixture = createTestComponent(html);
-    const compiled = fixture.nativeElement;
-    const buttonEl = compiled.querySelector('button');
+  describe('escape closing', () => {
+    const testEscapeClosing = ({autoClose, getElementForEventDispatch}) => {
```

Comment:
Personally, I would really much prefer reading 4 times duplicated code for many reasons:

- you have all you need in a single `it()` function, don't have to jump back and forth trying to understand what is passed where
- don't have to think when reading the tests (ex. conditions, remembering what were the arguments passed, reading `expectation.not.toBeShown()` vs `expect(compiled).not.toBeShown()`, etc.)
- all other tests (at least for this component) are written in this style

P.S. Would have nothing against this, if it was in the component code, not the test

@pkozlowski-opensource, should we change or leave like this?